init/devuan/walinuxagent

```diff
@@ -0,0 +1,605 @@
+#!/bin/bash
+# walinuxagent.sysvinit
+# sysvinit script to start and stop the waagent daemon.
+# 
+# This script takes into account the possibility that both daemon and
+# non-daemon instances of waagent may be running concurrently (e.g. when
+# "waagent -deprovision" is being run) and attempts to ensure that any
+# non-daemon instances are preserved when the daemon instance is stopped.
+# 
```

Comment:
is this the motivation for the checks on the pid file? i'm a little concerned about those because we are planning on re-structuring the agent and _may_ not need a pid file, and not having that file may affect this script.

in particular "waagent -deprovision" should not be executed while the agent service is running, since a running agent may re-create part of the state that deprovision is trying to delete,

Reply:
The new init script was based on observations whilst trying waagent on a local test VM - I noticed that using the existing init script, an attempt to stop waagent also killed an instance which was running in a terminal. I can't remember what the terminal instance was running - it wasn't a deprovision. The script was an attempt to cope with any foreseeable possibility - maybe it addresses possibilities which couldn't actually exist in practice.

I wondered about rewriting the script so that it doesn't rely on PID files, and works out which are the daemon and non-daemon instances just from the output of the ps command. That should insulate it against any possibility that the use of pidfiles could be dropped in future.

Reply:
I've just uploaded a new version of the devuan init script - I've rewritten it completely to remove all references to, and operations on, a pid file: it now operates only on the basis of output from the ps command.

Reply:
thanks for the update